truffle/src/main/java/org/jruby/truffle/translator/TranslatorDriver.java

```diff
@@ -167,7 +167,9 @@ private RubyRootNode parse(Node currentNode, RubyContext context, Source source,
 
         // Catch return
 
-        truffleNode = new CatchReturnAsErrorNode(context, truffleNode.getSourceSection(), truffleNode);
+        if (parserContext != ParserContext.EVAL) {
+            truffleNode = new CatchReturnAsErrorNode(context, truffleNode.getSourceSection(), truffleNode);
+        }
```

Comment:
Indeed, although ReturnID might get messed up if there is a `return` inside eval, right?